pkg/repo/mysql/services_integration_test.go

```diff
@@ -0,0 +1,224 @@
+package mysql
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"testing"
+
+	"github.com/brianvoe/gofakeit/v6"
+	"github.com/google/uuid"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/assert"
+	"go.expect.digital/translate/pkg/model"
+	"go.expect.digital/translate/pkg/repo"
+	"go.expect.digital/translate/pkg/tracer"
+)
+
+var mysqlRepo *Repo
+
+func TestMain(m *testing.M) {
+	ctx := context.Background()
+
+	viper.SetEnvPrefix("translate_mysql")
+	viper.AutomaticEnv()
+
+	tp, err := tracer.TracerProvider()
+	if err != nil {
+		log.Panicf("set tracer provider: %v", err)
+	}
+
+	conf := &Conf{
+		Host:     viper.GetString("host"),
+		Port:     viper.GetString("port"),
+		User:     viper.GetString("user"),
+		Database: viper.GetString("database"),
+	}
+
+	mysqlRepo, err = NewRepo(WithDBConfig(ctx, conf))
+	if err != nil {
+		log.Panicf("create new repo: %v", err)
+	}
+
+	code := m.Run()
+
+	mysqlRepo.db.Close()
+
+	if err := tp.Shutdown(ctx); err != nil {
+		log.Panicf("tp shutdown: %v", err)
+	}
+
+	os.Exit(code)
+}
+
+func createTestService() *model.Service {
+	return &model.Service{
+		Name: gofakeit.FirstName(),
+		ID:   uuid.New(),
+	}
+}
+
+// insertTestService inserts service to DB.
+func insertTestService(t *testing.T, ctx context.Context, service *model.Service) error {
```

Comment:
Doesn't `repository.SaveService()` do the same?

Reply:
Yes, but I thought that is not a good practice to use one repo function in the other.
If we use `repo.SaveService`, in all other tests "repo.LoadService" etc to insert data before testing actual functionality., then if SaveService does not work as intended then all tests are failing. So that makes them dependent on each other?

Reply:
As we climb higher and higher in the test pyramid, it becomes harder and harder to test. Keep it simple. If we have an implementation that does the job, we use it - we trust it, and it is tested as well. There is no point in duplicating implementation, now you have to worry about two implementations instead of one, and the effort of maintaining it grows exponentially. The system becomes more fragile.